### PR TITLE
hack/usoc22: Update basic-app-porting session

### PR DESCRIPTION
**Pull request by gabrielmocanu**
feat: Update basic-app-porting session

Signed-off-by: Gabriel Mocanu <gabi.mocanu98@gmail.com>

### CODE REVIEW

File: content/en/community/hackathons/sessions/basic-app-porting/content/Overview_Creating-a-Boilerplate-Microlibrary-for-Your-Application.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 To get started, we must create a new library for our application.
-The premise here is that we are going to "wrap" or "decorate" the source code of `iperf3` with the _lingua franca_ of Unikraft's build system.
+The premise here is that we are going to wrap or decorate the source code of `iperf3` with the _lingua franca_ of Unikraft's build system.
 That is, when we eventually build the application, the Unikraft build system will understand where to get the source code files, which ones to compile and how, with respect to the rest of Unikraft's internals and other dependencies.
```
Comment (StefanJum) [inline content/en/community/hackathons/sessions/basic-app-porting/content/Overview_Creating-a-Boilerplate-Microlibrary-for-Your-Application.md:3]:
```suggestion
That is, when we eventually build the application, the Unikraft build system will understand where to get the source code files from, which ones to compile and how, with respect to the rest of Unikraft's internals and other dependencies.
```

File: content/en/community/hackathons/sessions/basic-app-porting/content/Overview_Providing-Build-Files.md [hunk 2]
```diff
@@ -107,35 +105,35 @@ This process is usually very iterative because it requires building the unikerne
...
+   Because the application has been configured and we have fetched the contents, we can simply try running the build in the Unikraft application directory:
 
    ```bash
    $ cd ~/workspace/apps/iperf3
    $ kraft build
    ```
 
 1. (Optional) This step occurs less frequently, but is still useful to discuss in the context of porting an application to Unikraft.
-   Remember in [the Unikraft build lifecycle](#the-unikraft-build-lifecycle) that there is a step which occurs between fetching the remote origin code and compiling it. This step (3), known as `prepare`, is used to make modifications to the origin code before it is compiled.
+   Remember in [the Unikraft build lifecycle](#the-unikraft-build-lifecycle) that there is a step which occurs between fetching the remote original code and compiling it. This step (3), known as `prepare`, is used to make modifications to the origin code before it is compiled.
```
Comment (StefanJum) [inline content/en/community/hackathons/sessions/basic-app-porting/content/Overview_Providing-Build-Files.md:121]:
```suggestion
   Remember in [the Unikraft build lifecycle](community/hackathons/usoc22/basic-app-porting/#the-unikraft-build-lifecycle) that there is a step which occurs between fetching the remote original code and compiling it. This step (3), known as `prepare`, is used to make modifications to the origin code before it is compiled.
```

File: content/en/community/hackathons/sessions/basic-app-porting/content/Work-Items_Support-Files.md [hunk 3]
```diff
@@ -1,24 +1,24 @@
-Session support files are available [in the repository](https://github.com/unikraft/summer-of-code-2021).
+Session support files are available [in the repository](https://github.com/unikraft/docs).
 If you already cloned the repository, update it and enter the session directory:
 
 ```
```
Comment (StefanJum) [inline content/en/community/hackathons/sessions/basic-app-porting/content/Work-Items_Support-Files.md:4]:
```suggestion
```bash
```

File: content/en/community/hackathons/sessions/basic-app-porting/content/Work-Items_Support-Files.md [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
 
-$ ls -F
-index.md  sol/  unikraft-overview.svg  work/
+$ ls
+index.md  sol/  unikraft-overview.svg  work/  content/
 ```
 
 If you haven't cloned the repository yet, clone it and enter the session directory:
 
 ```
```
Comment (StefanJum) [inline content/en/community/hackathons/sessions/basic-app-porting/content/Work-Items_Support-Files.md:17]:
```suggestion
```bash
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Some comments that I cannot add to the `files` section, since there were no changes to those files:

* [Here](https://github.com/unikraft/docs/blob/717dd24aeab1aa7feff3450411f8e599dedb1f64/content/en/community/hackathons/sessions/basic-app-porting/content/Overview_Identifying-a-Candidate-Application.md?plain=1#L9), instead of `C/++-based` --> `C/C++-based`.
* I would add some example links [here](https://github.com/unikraft/docs/blob/717dd24aeab1aa7feff3450411f8e599dedb1f64/content/en/community/hackathons/sessions/basic-app-porting/content/Overview_Identifying-a-Candidate-Application.md?plain=1#L15) to [app-python](https://github.com/unikraft/app-python3/), [app-lua](https://github.com/unikraft/app-lua), etc.
* Add `bash` for the code snippets [here](https://github.com/unikraft/docs/blob/717dd24aeab1aa7feff3450411f8e599dedb1f64/content/en/community/hackathons/sessions/basic-app-porting/content/Overview_Using-Your-Library-in-a-Unikraft-Unikernel-Application.md?plain=1#L7).
* [This](https://github.com/unikraft/docs/blob/717dd24aeab1aa7feff3450411f8e599dedb1f64/content/en/community/hackathons/sessions/basic-app-porting/content/Overview_Patching-the-Application.md?plain=1#L31) is outdated since now we have some stub implementation for `mmap` and a stub for `munmap`, so the patch should not be necessary, but I guess we can keep this patch and add a note, just to keep things simple but updated.
* [This](https://github.com/unikraft/docs/blob/717dd24aeab1aa7feff3450411f8e599dedb1f64/content/en/community/hackathons/sessions/basic-app-porting/content/Work-Items_01.-Port-%60libfortune%60-to-Unikraft.md?plain=1#L11-L33) should be updated, both the GitHub links, paths, and the Unikraft banner version.

**Comment by gabrielmocanu** [comment]
1. Done
3. Done
4. Done

**Comment by StefanJum** [comment]
@gabrielmocanu it looks good now, besides the problem with the lib that you are working on and the quiz we will need to have.

**Review by StefanJum — APPROVED** [review]
Looks good now, we will update the work items in a later pr.
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**